lisa/node.py

```diff
@@ -531,7 +532,26 @@ def _initialize(self, *args: Any, **kwargs: Any) -> None:
             )
             self._first_initialize = True
         self.log.info(f"initializing node '{self.name}' {self}")
-        self.shell.initialize()
+        try:
+            self.shell.initialize()
+        except TcpConnectionException as e:
+            from lisa.features import SerialConsole
+
+            if self.features.is_supported(SerialConsole):
+                serial_console = self.features[SerialConsole]
+
+                # @TODO: this is temporary, we need another way to determine the commands.
+                # OS is not initialized yet, so cannot use self.os.
+                commands = [
+                    "ip addr show",
+                    "ip link show",
+                    "systemctl status NetworkManager --no-pager --plain",
+                    "systemctl status network --no-pager --plain",
+                    "systemctl status systemd-networkd --no-pager --plain",
+                    "ping -c 3 -n 8.8.8.8",
+                ]
+                serial_console.execute_command(commands=commands)
```

Comment:
I'm not sure how much this will help, but here are a couple of points:
Most VMs appear to be stuck at the login screen, so it's not possible to input any commands directly.
We need to wait until the command produces output. If the case fails quickly, the downloaded serial log might not contain some keywords. Please try using get_console_log repeatedly until the serial log has content or a timeout occurs.

Reply:
Thanks for the comment, Chi.
I will handle the read-retry-timeout scenario.

What do you think is the best way to determine the list of commands to execute?

> I'm not sure how much this will help 
The reason I feel this would help is because, right now when LISA fails with ssh, we have no insight into what is happening in the VM. If we can craft a list of commands that would collection sufficient log in most scenarios, that itself would be a great reduction in the need for a repro

Reply:
Did you try it in real cases? Especially, the serial console is not signed in.

Reply:
I haven't tried the stuck at login prompt scenario yet.
So far I have only tried in scenario where the VM was booted and running, but it had issues with inbound connection.